### PR TITLE
Detach config

### PR DESCRIPTION
**Pull request by psmolkin**
Hi!
Thanks for a great job firstly!
My little little improvments for this project:

* Detach config and make it auto-creatable
* Disabled loading PS profile to speed up startup

### CODE REVIEW

File: distro/scripts/startup.sh [hunk 1]
```diff
@@ -1,11 +1,17 @@
 #! /bin/sh
 
 LOG_PATH="/var/log/wsl-vpnkit.log"
-USERPROFILE=$(wslpath "$(powershell.exe -c 'Write-Host -NoNewline $env:USERPROFILE')")
+USERPROFILE=$(wslpath "$(powershell.exe -NoLogo -NoProfile -c 'Write-Host -NoNewline $env:USERPROFILE')")
```
Comment (sakai135) [inline distro/scripts/startup.sh:4]: TIL, having the `-NoProfile` is a great improvement that I missed

File: distro/scripts/startup.sh [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
 VERSION="$(cat /app/version)"
+CONF_PATH="$USERPROFILE/wsl-vpnkit/wsl-vpnkit.conf"
 
 touch $LOG_PATH
 
+mkdir -p "$USERPROFILE/wsl-vpnkit/"
+if [ ! -f "$CONF_PATH" ]; then
+  cp /app/defaults.conf "$CONF_PATH"
+fi
```
Comment (sakai135) [inline distro/scripts/startup.sh:13]: To minimize the changes automatically made by the script to the user's machine, I would rather this to be added as part of the startup text so users could manually create the config file. That will remove the need to care about migrating the config if something ever changes.

File: distro/scripts/startup.sh [hunk 3]
```diff
@@ -15,6 +21,13 @@ Run the following commands from Windows or other WSL 2 distros to use.
     wsl.exe -d $WSL_DISTRO_NAME service wsl-vpnkit start
     wsl.exe -d $WSL_DISTRO_NAME service wsl-vpnkit stop
 
+or just create an alias:
+
+    echo 'alias vpnkit="wsl.exe -d wsl-vpnkit service wsl-vpnkit"' >/etc/profile.d/vpnkit.sh
+    source /etc/profile.d/vpnkit.sh
+    vpnkit start
+    vpnkit stop
```
Comment (sakai135) [inline distro/scripts/startup.sh:29]: This script could vary between different distros, so I would rather omit this and leave the users to create their own setup.

File: distro/scripts/startup.sh (hunk 2 above)
Comment (psmolkin) [reply distro/scripts/startup.sh]: Agree. The main motivation was to separate the evaluated variables from the constants.

**Comment by psmolkin** [comment]
Updated as per comments above

File: wsl-vpnkit [hunk 4]
```diff
@@ -11,24 +11,30 @@ VPNKIT_STORE="/files/vpnkit/vpnkit.exe"
...
-VPNKIT_DEBUG=
-WSL2_GATEWAY_IP="$(cat /etc/resolv.conf | awk '/^nameserver/ {print $2}')"
-CHECK_HOST=example.com
-DNS_IP="$VPNKIT_GATEWAY_IP"
-CHECK_DNS=1.1.1.1
 
 echo "starting wsl-vpnkit"
+
+# Load defaults and user config if needed
+. /app/defaults.conf
```
Comment (sakai135) [inline wsl-vpnkit:18]: We should do a check if this default config file exists before trying to load it.

Comment (psmolkin) [reply wsl-vpnkit]: ✔

**Comment by psmolkin** [comment]
What do you think about change `powershell.exe` call to `cmd.exe`?
```sh
time -f '%e' ash -c "wslpath \$(cmd.exe /c 'echo %USERPROFILE%' 2>/dev/null)"
/mnt/c/Users/smolkinpv
0.09

time -f '%e' ash -c "wslpath \$(powershell.exe -NoLogo -NoProfile -c 'Write-Host -NoNewline \$env:USERPROFILE')"
/mnt/c/Users/smolkinpv
0.96
```

**Comment by psmolkin** [comment]
as well as
```sh
time -f '%e' taskkill.exe /F /T /IM wsl-vpnkit.exe || :
SUCCESS: The process with PID 9072 (child process of PID 10816) has been terminated.
0.42
time -f '%e' powershell.exe -NoProfile -c 'Stop-Process -Force -Name wsl-vpnkit -ErrorAction SilentlyContinue'
0.97
```

**Comment by sakai135** [comment]
@psmolkin Thanks, those improvements do seem worthwhile too if you want to open a separate PR.

**Pull request merged**